src/main/java/soonkeatneo/duke/TaskList.java

```diff
@@ -0,0 +1,186 @@
+     * @param inputString the String to search for
+     */
+    public void find(String inputString) {
+        List<Task> newList = new ArrayList<>();
+        boolean hasPrintedTasks = false;
+        int counter = 0;
+        for (int i = 0; i < this.taskList.size(); i++) {
+            Task task = this.taskList.get(i);
+            if (task.getDescription().toLowerCase().contains(inputString.toLowerCase())) {
+                if (!hasPrintedTasks) {
+                    Ui.printMessage("Here are the list of matching tasks: ");
+                    hasPrintedTasks = true;
+                }
+                Ui.printMessage((counter + 1) + ". " + task.toString());
+                counter += 1;
+            }
+        }
+        if (!hasPrintedTasks) {
+            Ui.printMessage("There are no matching tasks!");
+        }
+    }
+
+    /**
+     * Adds a new {@Task} to the task list.
+     * @param task Given task to be added to the task list
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+        Ui.printMessage("Wakarimashita! Task added to list:");
+        Ui.printMessage(task.toString());
+        Ui.printMessage("The size of your task list is now: " + this.getSize());
+    }
+
+    /**
+     * Mark the specified task as complete.
+     * @param inputString User input string
+     */
+    public void completeTask(String inputString, Storage storage) {
+        try {
+            int taskId = Integer.parseInt(String.valueOf(inputString.split(" ")[1])) - 1;
+            Task doneTask = taskList.get(taskId).setDone();
+            String typeOfTask = doneTask.getType();
+            String completionOfTask = (doneTask.getDone() ? "1" : "0");
+            String descriptionOfTask = doneTask.getDescription().strip();
+            LocalDate date = LocalDate.now();
+            if (doneTask instanceof Event) {
+                date = ((Event) doneTask).getDate();
+            } else if (doneTask instanceof Deadline) {
+                date = ((Deadline) doneTask).getDate();
+            }
+            String oldString = typeOfTask + " | " + "0" + " | " + descriptionOfTask;
+            String newString = typeOfTask + " | " + completionOfTask + " | " + descriptionOfTask;
+            if (doneTask instanceof Event || doneTask instanceof Deadline) {
+                oldString += " | " + date.toString();
+                newString += " | " + date.toString();
+            }
+            storage.deleteReplaceTaskFromDisk(oldString, newString);
+            Ui.printMessage("Great~! Task completed:");
+            Ui.printMessage(doneTask.toString());
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidTaskException();
+        }
+    }
+
+    /**
+     * Delete the specified task from the list.
+     * @param inputString User input string
+     * @param storage Storage object to remove from disk
+     */
+    public void deleteTask(String inputString, Storage storage) {
```

Comment:
Do you think the `deleteTask` and `completeTask` share some similarities here? What about abstracting something out? 🤔